zuko/flows/autoregressive.py

```diff
@@ -120,21 +122,31 @@ def __init__(
             self.order = torch.div(order, ceil(features / self.passes), rounding_mode="floor")
 
             adjacency = self.order[:, None] > self.order
+            context_adjacency = None
         else:
             adjacency = torch.as_tensor(adjacency, dtype=bool)
 
             assert adjacency.ndim == 2, "'adjacency' should be a matrix."
             assert adjacency.shape[0] == features, f"'adjacency' should have {features} rows."
-            assert adjacency.shape[1] == features, f"'adjacency' should have {features} columns."
+            assert adjacency.shape[1] == features or adjacency.shape[1] == features + context, (
+                f"'adjacency' should have {features} or {features + context} columns."
+            )
+
+            has_context = adjacency.shape[1] == features + context
+            context_adjacency = adjacency[:, features:] if has_context else None
+            adjacency = adjacency[:, :features]
+
             assert adjacency.diag().all(), "'adjacency' should have ones on the diagonal."
 
             adjacency = adjacency * ~torch.eye(features, dtype=bool)
 
             self.passes = self._dag_diameter(adjacency)
 
         if context > 0:
+            if context_adjacency is None:
+                context_adjacency = torch.ones((features, context), dtype=bool)
             adjacency = torch.cat(
-                (adjacency, torch.ones((features, context), dtype=bool)),
+                (adjacency, context_adjacency),
```

Comment:
```suggestion
                (adjacency, adjacency_context),
```

Could you also make this operation a single line?

Reply:
Sure!